lucetc/src/compiler.rs

```diff
@@ -308,6 +321,113 @@ impl<'a> Compiler<'a> {
             function_map.insert(func_id, (size, trap_data_id, traps.len()));
         }
 
+        // Now that we've defined all functions, we know what trampolines must also be created.
+        //
+        // Hostcall trampolines have the general shape of:
+        //
+        // ```
+        // fn trampoline_$hostcall(&vmctx, $hostcall_args) -> $hostcall_result {
+        //     if context.rsp < vmctx.instance_implicits.stack_limit {
+        //         // insufficient stack space to make the call
+        //         terminate_with_stack_overflow();
+        //     }
+        //
+        //     $hostcall(vmctx, $hostcall_args..)
+        // }
+        // ```
+        //
+        // but are specified here as Cranelift IR for lack of source to generate them from.
+        for (hostcall_name, (trampoline_id, hostcall_func_index)) in &self.trampolines {
+            let mut trampoline_context = ClifContext::new();
```

Comment:
the enclosing function is getting a bit big, I'm planning on taking this loop body and moving it to a `fn synthesize_trampoline(trampoline_id, hostcall_func_index)`